### PR TITLE
feat: add environment variable substition to source queries

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

Adds interpolation for variables in source queries using `${}` syntax

"Advanced" Connectors have the option to disable this functionality when reading files (it is done automatically for "simple" connectors)

All variables must be in the environment as `EVIDENCE_VAR__[var name]` to ensure that they were deliberately set.

e.g.

```sql
SELECT * FROM MY_TABLE
WHERE client_id = ${client_id}
```

```bash
EVIDENCE_VAR__client_id=5 pnpm run sources
```

```sql
-- actually runs
SELECT * FROM MY_TABLE
WHERE client_id = 5
```

### Checklist

~~- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after~~
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by archiewood** [comment]
We may consider scoping this syntax?

For example:


```sql
SELECT * FROM MY_TABLE
WHERE client_id = ${env.client_id}
```

or 

```sql
SELECT * FROM MY_TABLE
WHERE client_id = ${var.client_id}
```


This might make it easier to distinguish variables from other queries if we were to add query chaining in the future?

```sql
SELECT * FROM ${my_other_query}
WHERE client_id = ${env.client_id}
```

**Comment by ItsMeBrianD** [comment]
> This might make it easier to distinguish variables from other queries if we were to add query chaining in the future?

I think this may merit a larger discussion; query chaining will have to be (or rather, really should be I think) implemented at the connector level, rather than by Evidence itself, because the rules for doing so will vary based on the query syntax (e.g. how would query chaining work for yaml-based queries). Simple raw variable interpolation is safer and should be pretty portable.

**Comment by archiewood** [comment]
Sounds good!

File: packages/plugin-connector/src/data-sources/sub-source-vars.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+				const name = k.substring(envPrefix.length);
+				const value = v?.toString();
+				return [name, value];
+			})
+	);
+	let output = queryString;
+	const regex = RegExp(/\$\{(.+?)\}/, 'g');
+	let match;
+	while ((match = regex.exec(queryString)) !== null) {
+		if (match[1] in validVars) output = output.replace(match[0], validVars[match[1]]);
```
Comment (archiewood) [inline packages/plugin-connector/src/data-sources/sub-source-vars.js:31]: This appears to throw
> │ > tsc --noEmit -p tsconfig.json && parcel build
│ src/data-sources/sub-source-vars.js(33,64): error TS2769: No overload matches this call.
│   The last overload gave the following error.
│     Argument of type 'string | undefined' is not assignable to parameter of type '(substring: s…
│       Type 'undefined' is not assignable to type '(substring: string, ...args: any[]) => string…

File: packages/plugin-connector/src/data-sources/sub-source-vars.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		else
+			console.warn(
+				`Missed substition for ${match[0]}, do you need to set EVIDENCE_VAR__${match[1]}?`
+			);
+	}
+
+	return queryString;
```
Comment (archiewood) [inline packages/plugin-connector/src/data-sources/sub-source-vars.js:38]: Should this return `output`?

**Comment by ItsMeBrianD** [comment]
Added a manual escape for users as well; `${variable}` gets replaced with the variable value, `$${variable}` gets replaced with `${variable}`

**Review by archiewood — APPROVED** [review]
Tested, works great for me!

**Pull request merged**